File: frontend/src/interfaces/Payment.ts
import { UserInterface } from './User'

export interface Payment {

  Date: string; // วันที่ (ISO 8601 format, required)

  UserID: number; // User ID
  User?: UserInterface; // Optional, ข้อมูลผู้ใช้ที่เกี่ยวข้อง

  PaymentMethodID: number;
  PaymentMethod: PaymentMethod;

  PaymentStatusID: number;
  PaymentStatus: PaymentStatus;

}

export interface PaymentMethod {

  PaymentMethod: string;

}

export interface PaymentStatus {

  PaymentStatus: string;

}

File: backend/entity/Payment.go
package entity

import (
	"gorm.io/gorm"
	"time"
)

type Payment struct {
	gorm.Model
	
	Date				time.Time		`valid:"required~Date is required"`

	UserID				uint 			`valid:"required~UserID is required"`
	User   				*Users 			`gorm:"foreignKey:UserID"`
	
	Order 				[]Order 		`gorm:"foreignKey:PaymentID"`

	PaymentMethodID		uint 			`valid:"required~PaymentMethodID is required"`
	PaymentMethod   	PaymentMethod 	`gorm:"foreignKey:PaymentMethodID"`

	PaymentStatusID		uint 			`valid:"required~PaymentStatusID is required"`
	PaymentStatus   	PaymentStatus	`gorm:"foreignKey:PaymentStatusID"`

	AddressID			uint 			`valid:"required~AddressID is required"`
	Address   			Address			`gorm:"foreignKey:AddressID"`
}


